### PR TITLE
Added normalized surface dice score in the BraTS evaluation metrics

### LINKED ISSUE #380: Add Normalized Surface DICE Score in the evaluation metrics
For the Evaluation MLCube, we're going to follow the implementation for the FeTS challenge, plus adding a new metric, the Normalized Surface DICE Score

### PR DESCRIPTION
**Pull request by sarthakpati**
Fixes #380

### CODE REVIEW

File: examples/BraTS/metrics/project/normalized_surface_dice.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+    prediction_img = sitk.ReadImage(input_prediction)
+    target_img = sitk.ReadImage(input_target)
+
+    assert (
+        prediction_img.GetSize() == target_img.GetSize()
+    ), "The prediction and target images must have the same size."
+    assert (
+        prediction_img.GetSpacing() == target_img.GetSpacing()
+    ), "The prediction and target images must have the same spacing."
+    prediction = torch.from_numpy(sitk.GetArrayFromImage(prediction_img)).long()
```
Comment (vpchung) [inline examples/BraTS/metrics/project/normalized_surface_dice.py:177]:
```suggestion
    prediction = torch.from_numpy(sitk.GetArrayFromImage(prediction_img).astype(np.int16)).long()
```

File: examples/BraTS/metrics/project/normalized_surface_dice.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    target = torch.from_numpy(sitk.GetArrayFromImage(target_img)).long()
```
Comment (vpchung) [inline examples/BraTS/metrics/project/normalized_surface_dice.py:178]:
```suggestion
    target = torch.from_numpy(sitk.GetArrayFromImage(target_img).astype(np.int16)).long()
```

File: examples/BraTS/metrics/project/app.py [hunk 3]
```diff
@@ -9,6 +9,8 @@
 
 import pandas as pd
 
+from .normalized_surface_dice import compute_surface_dice_from_images
```
Comment (vpchung) [inline examples/BraTS/metrics/project/app.py:12]:
```suggestion
from normalized_surface_dice import compute_surface_dice_from_images
```

**Review by vpchung — CHANGES_REQUESTED** [review]
We addressed these during the call today, but I'm going to submit them just to keep notes :-)

File: examples/BraTS/metrics/project/normalized_surface_dice.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (vpchung) [reply examples/BraTS/metrics/project/normalized_surface_dice.py:177]: We've identified that a potential issue could be from the version of SimpleITK.  A possible fix would be to exclude certain versions in requirements.txt, e.g.

```
PyYAML~=5.3
pandas~=1.1
typer
MedPy
SimpleITK!=2.0.*,!=2.2.1
numpy
torch
```

**Comment by vpchung** [comment]
@sarthakpati I just submitted a CLA sign up form.

**Review by vpchung — COMMENTED** [review]
All looks good on my end.

I do have a quick question about how the participants should submit their predictions though -- it's a little different from what we asked for in BraTS2021 and BraTS2022.

**Comment by vpchung** [comment]
CLA has been signed.

**Review by aristizabal95 — APPROVED** [review]
Looks good on medperf's side. @vpchung verified the implementation as well

**Pull request merged**